File: upl-purchase-sdk-java/src/main/java/com/unzer/paylater/model/RefundReason.java
/**
 * Purchase API
 * @copyright Copyright (c) 2020 Unzer Pay Later
 * @license see LICENSE.TXT
 *
 * This class is based on the Unzer Pay Later OpenAPI specification, version 1.0.0.
 */
package com.unzer.paylater.model;

public enum RefundReason {

    CUSTOMER_REFUND("CUSTOMER_REFUND"),
    MERCHANT_TECHNICAL_PROBLEM("MERCHANT_TECHNICAL_PROBLEM"),
    REFUND_OBLIGINGNESS("REFUND_OBLIGINGNESS"),
    MERCHANT_CAN_NOT_DELIVER_GOODS("MERCHANT_CAN_NOT_DELIVER_GOODS");

    private String value;

    RefundReason(String value) {
        this.value = value;
    }

    public static RefundReason fromValue(String value) {
        for (RefundReason b : RefundReason.values()) {
            if (b.value.equals(value)) {
                return b;
            }
        }
        throw new IllegalArgumentException("Unexpected value '" + value + "'");
    }

    public String getValue() {
        return value;
    }

    @Override
    public String toString() {
        return String.valueOf(value);
    }
}

File: upl-purchase-sdk-php/src/Model/RefundReason.php
<?php

/**
 * Purchase API
 * @copyright Copyright (c) 2020 Unzer Pay Later
 * @license see LICENSE.TXT
 *
 * This class is based on the Unzer Pay Later OpenAPI specification, version 1.0.0.
 */

declare(strict_types=1);

namespace Unzer\PayLater\Model;

use InvalidArgumentException;

use function array_key_exists;
use function sprintf;

class RefundReason implements Enum
{
    public const CUSTOMER_REFUND = 'CUSTOMER_REFUND';
    public const MERCHANT_TECHNICAL_PROBLEM = 'MERCHANT_TECHNICAL_PROBLEM';
    public const REFUND_OBLIGINGNESS = 'REFUND_OBLIGINGNESS';
    public const MERCHANT_CAN_NOT_DELIVER_GOODS = 'MERCHANT_CAN_NOT_DELIVER_GOODS';

    public const ALLOWED_TYPES = [
        'CUSTOMER_REFUND' => self::CUSTOMER_REFUND,
        'MERCHANT_TECHNICAL_PROBLEM' => self::MERCHANT_TECHNICAL_PROBLEM,
        'REFUND_OBLIGINGNESS' => self::REFUND_OBLIGINGNESS,
        'MERCHANT_CAN_NOT_DELIVER_GOODS' => self::MERCHANT_CAN_NOT_DELIVER_GOODS,
    ];

    /**
     * @var string
     */
    private $value;

    /**
      * @param string $value
      * @throws InvalidArgumentException
      */
    public function __construct(string $value)
    {
        if (!array_key_exists($value, self::ALLOWED_TYPES)) {
            throw new InvalidArgumentException(sprintf('Unexpected value "%s"', $value));
        }
        $this->value = $value;
    }

    /**
     * @return string
     */
    public function getValue(): string
    {
        return $this->value;
    }

    /**
     * @return string
     */
    public function __toString(): string
    {
        return $this->value;
    }
}
